src/backend/config.py

```diff
@@ -54,17 +55,32 @@ def get_api_path(ui_path: str) -> str:
     return f"{ui_path.rstrip('/')}/api"
 
 
+def get_s3_addressing_style() -> Literal["auto", "virtual", "path"]:
+    """
+    Obtain the S3 addressing style. Defaults to `auto`.
+    """
+    addressing_style: Literal["auto", "virtual", "path"] = "auto"
+    env_val = os.environ.get("S3GW_S3_ADDRESSING_STYLE")
```

Comment:
just for good measure, I think we should do a `lower()` here.

Also, I would declare `env_val` as `str | None` -- I think that may resolve the General Type Issue below.

Reply:
Done